src/peer_store.rs

```diff
@@ -32,6 +32,10 @@ where
 	pub(crate) fn add_peer(&self, peer_info: PeerInfo) -> Result<(), Error> {
 		let mut locked_peers = self.peers.write().unwrap();
 
+		if locked_peers.contains_key(&peer_info.node_id) {
+			return Ok(());
+		}
```

Comment:
Could you make this a separate commit stating why this is needed? Assuming this is needed regardless of the larger change.